app/scripts/components/analysis/results/index.tsx

```diff
@@ -133,16 +133,28 @@ export default function AnalysisResults() {
 
   const availableDomain: [Date, Date] | null = useMemo(() => {
     if (!start || !end) return null;
+    let minDate = +start;
+    let maxDate = +end;
+    requestStatus.forEach((item) => {
+      if (item.data?.timeseries) {
+        const itemDates = item.data.timeseries.map((t) => +new Date(t.date));
+        const itemMin = min(itemDates) ?? Number.POSITIVE_INFINITY;
+        const itemMax = max(itemDates) ?? Number.NEGATIVE_INFINITY;
+        if (itemMin < minDate) minDate = itemMin;
+        if (itemMax > maxDate) maxDate = itemMax;
+      }
+    });
```

Comment:
@nerik Thoughts on using reduce to not mutate values?

```suggestion

    const { minDate, maxDate } = requestStatus.reduce(
      (acc, item) => {
        if (item.data?.timeseries) {
          const itemDates = item.data.timeseries.map((t) => +new Date(t.date));
          const itemMin = min(itemDates) ?? Number.POSITIVE_INFINITY;
          const itemMax = max(itemDates) ?? Number.NEGATIVE_INFINITY;

          return {
            minDate: itemMin < acc.minDate ? itemMin : acc.minDate,
            maxDate: itemMax > acc.maxDate ? itemMax : acc.maxDate
          };
        }
        return acc;
      },
      { minDate: +start, maxDate: +end }
    );
```

Reply:
In that specific case (correct me if I'm wrong), there are no mutating values as minDate and maxDate are primitive data types and thus immutable. I avoid using `reduce` when not forced to, as it is slightly harder to reason about than other looping structures, but it's really down to personal preference. Maybe it's good to make a habit of using reduce for the majority of cases where you'd actually be mutating something. 

You have the final word.
![](https://media1.giphy.com/media/H3N0cu89FUOaB5CTbv/giphy.gif?cid=ecf05e472ax2wud79l0nisn92i6hk66wghof6u33cjddhrhq&rid=giphy.gif&ct=g)

Reply:
You're right, nothing is mutating just re-assigning values. I tend to avoid this since it may be really hard to know how may times things get changed, and I like to consistently not do it unless necessary. But then again, this is such a small case that it really is matter of personal preference.
Will update just to keep consistent with other parts of the app.